### PR TITLE
remove olympus storage implementation and proxy worker

### PR DESCRIPTION
**Pull request by rchakra3**
Addresses #414 

The only worker in the proxy was set up to asynchronously pull from olympus on cache misses. I've deleted all that code in addition to the olympus storage module. As mentioned in #414 if we want to bring back the async download functionality, we can implement a new `download.Protocol` for olympus

### CODE REVIEW

**Comment by michalpristas** [comment]
please resolve conflicts @rchakra3

File: cmd/proxy/actions/filter.go [hunk 1]
```diff
@@ -7,11 +7,24 @@ import (
...
 	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/config/env"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/module"
 	"github.com/gomods/athens/pkg/paths"
 )
 
+const (
+	// OlympusGlobalEndpoint is a default olympus DNS address
+	OlympusGlobalEndpoint = "http://localhost:3001"
```
Comment (marwan-at-work) [inline cmd/proxy/actions/filter.go]: Is this file changed by mistake? this seems out of scope (and already addressed in another PR if im not mistaken)

**Comment by rchakra3** [comment]
@marwan-at-work I think your comments aren't applicable after moving the middleware to pkg/

@michalpristas resolved conflicts!

File: cmd/proxy/actions/app.go [hunk 2]
```diff
@@ -60,8 +44,8 @@ func init() {
 // should be defined. This is the nerve center of your
 // application.
 func App() (*buffalo.App, error) {
-	ctx := context.Background()
 	store, err := GetStorage()
+
```
Comment (michalpristas) [inline cmd/proxy/actions/app.go]: can you please remove space?

**Comment by michalpristas** [comment]
@marwan-at-work can you please review once more. this is a quick one
